File: backend/Recodo.Common/Dtos/Folder/FolderDTO.cs
using System.Collections.Generic;

namespace Recodo.Common.Dtos.Folder
{
    public class FolderDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int AuthorId { get; set; }
        public int? ParentId { get; set; }
        public int? TeamId { get; set; }
    }
}

File: frontend/src/app/models/folder/folder-dto.ts
import { UserDto } from '../user/user-dto';

export interface FolderDto{
  id:number;
  name:string;
  authorId:number;
  author:UserDto;
  parent:FolderDto;
  parentId:number;
  teamId:number;
  subFolders:FolderDto[];
}